### ISSUE TITLE
empty notes results in a stack strace in iodef lib

### ISSUE THREAD

**Issue by kruisdraad** [AbuseIO/AbuseIO#201]
```
Jan  2 09:00:44 abuseio abuseio[13969]: production.ERROR: ErrorException: Array to string conversion in /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php:269 Stack trace: #0 /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php(269): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8, 'Array to string...', '/opt/abuseio/ve...', 269, Array) #1 /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php(218): Marknl\Iodef\IodefElement->validate() #2 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(174): Marknl\Iodef\IodefElement->xmlSerialize(Object(Marknl\Iodef\Writer)) #3 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\AdditionalData)) #4 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(201): Sabre\Xml\Writer->write(Object(Marknl\Iodef\Elements\AdditionalData)) #5 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(212): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\AdditionalData)) #6 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Array) #7 /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php(225): Sabre\Xml\Writer->write(Array) #8 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(174): Marknl\Iodef\IodefElement->xmlSerialize(Object(Marknl\Iodef\Writer)) #9 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\HistoryItem)) #10 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(201): Sabre\Xml\Writer->write(Object(Marknl\Iodef\Elements\HistoryItem)) #11 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(212): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\HistoryItem)) #12 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Array) #13 /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php(225): Sabre\Xml\Writer->write(Array) #14 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(174): Marknl\Iodef\IodefElement->xmlSerialize(Object(Marknl\Iodef\Writer)) #15 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\History)) #16 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(201): Sabre\Xml\Writer->write(Object(Marknl\Iodef\Elements\History)) #17 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(212): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\History)) #18 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Array) #19 /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php(225): Sabre\Xml\Writer->write(Array) #20 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(174): Marknl\Iodef\IodefElement->xmlSerialize(Object(Marknl\Iodef\Writer)) #21 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\Incident)) #22 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(201): Sabre\Xml\Writer->write(Object(Marknl\Iodef\Elements\Incident)) #23 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(212): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\Incident)) #24 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Array) #25 /opt/abuseio/vendor/marknl/iodef/src/IodefElement.php(225): Sabre\Xml\Writer->write(Array) #26 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(174): Marknl\Iodef\IodefElement->xmlSerialize(Object(Marknl\Iodef\Writer)) #27 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\IODEFDocument)) #28 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(201): Sabre\Xml\Writer->write(Object(Marknl\Iodef\Elements\IODEFDocument)) #29 /opt/abuseio/vendor/sabre/xml/lib/Serializer/functions.php(212): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Object(Marknl\Iodef\Elements\IODEFDocument)) #30 /opt/abuseio/vendor/sabre/xml/lib/Writer.php(100): Sabre\Xml\Serializer\standardSerializer(Object(Marknl\Iodef\Writer), Array) #31 /opt/abuseio/vendor/abuseio/notification-mail/src/Mail.php(140): Sabre\Xml\Writer->write(Array) #32 /opt/abuseio/app/Jobs/Notification.php(44): AbuseIO\Notification\Mail->send(Array) #33 /opt/abuseio/app/Jobs/Notification.php(111): AbuseIO\Jobs\Notification->send(Array) #34 /opt/abuseio/app/Console/Commands/Housekeeper/NotificationsCommand.php(141): AbuseIO\Jobs\Notification->walkList(Array) #35 [internal function]: AbuseIO\Console\Commands\Housekeeper\NotificationsCommand->handle() #36 /opt/abuseio/vendor/laravel/framework/src/Illuminate/Container/Container.php(507): call_user_func_array(Array, Array) #37 /opt/abuseio/vendor/laravel/framework/src/Illuminate/Console/Command.php(150): Illuminate\Container\Container->call(Array) #38 /opt/abuseio/vendor/symfony/console/Command/Command.php(254): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #39 /opt/abuseio/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #40 /opt/abuseio/vendor/symfony/console/Application.php(847): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #41 /opt/abuseio/vendor/symfony/console/Application.php(190): Symfony\Component\Console\Application->doRunCommand(Object(AbuseIO\Console\Commands\Housekeeper\NotificationsCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #42 /opt/abuseio/vendor/symfony/console/Application.php(121): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #43 /opt/abuseio/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(107): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #44 /opt/abuseio/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #45 {main}
```

Object that resulted in this error:

```
    object(Marknl\Iodef\Elements\AdditionalData)#6818 (4) {
      ["value"]=>
      string(1) " "
      ["ns":protected]=>
      string(34) "{urn:ietf:params:xml:ns:iodef-1.0}"
      ["attributes":protected]=>
      array(5) {
        ["dtype"]=>
        string(6) "string"
        ["ext-dtype"]=>
        string(0) ""
        ["meaning"]=>
        string(4) "text"
        ["formatid"]=>
        string(0) ""
        ["restriction"]=>
        string(0) ""
      }
      ["elements":protected]=>
      array(0) {
      }
    }
```

fixed with: 
notification-mail/src/Mail.php:

501,506c501
< 
<                 if(empty(trim($note->text))) {
<                     $historyNote->value = false;
<                 } else {
<                     $historyNote->value = "{$note->text}";
<                 }

**Comment by kruisdraad** [comment]
This needs to be patches in both 1.2.x and 1.3.x versions

**Comment by kruisdraad** [comment]
fixed with commit in master 1.3 and branch 1.2.x
